js/render/scenes/scene.js

```diff
@@ -229,53 +229,70 @@ export class Scene extends Node {
   }
 
   /** Draws the scene into the base layer of the XRFrame's session */
-  drawXRFrame(xrFrame, pose) {
+  drawXRFrame(xrFrame, pose, layer, depthData) {
     if (!this._renderer || !pose) {
       return;
     }
 
     let gl = this._renderer.gl;
+    let renderer = this._renderer;
     let session = xrFrame.session;
     // Assumed to be a XRWebGLLayer for now.
-    let layer = session.renderState.baseLayer;
-    if (!layer)
-      layer = session.renderState.layers[0];
-    else {
+    let xrlayer = session.renderState.baseLayer;
+    let views = [];
+
+    if (!xrlayer) {
+      xrlayer = layer ? layer : session.renderState.layers[0];
+      gl.bindFramebuffer(gl.FRAMEBUFFER, renderer.xrFramebuffer);
+
+      for (let view of pose.views) {
+        let glLayer = renderer.getXrBinding(session).getViewSubImage(xrlayer, view);  
```

Comment:
I think this code relies on Layers? How does it handle if they aren't present/enabled?

Reply:
I'm assuming that browser that support multiview also support layers since multiview is useless with the regular xrwebgllayer. 
Do you know a browser that supports multiview apart from quest?

Reply:
I'm not seeing where this has checked for multiview? Further, this glLayer seems to be used even in the final else of the chain that starts on 252?

Reply:
I'm not following. If we get here, we know that the session is using layers. The decision to use multiview come from the renderer and is used in line 252-255

Reply:
So, to my eyes this looks like a generic `scene` class that I would think all of the samples are or could be using, and nothing between the `drawXRFrame` function declaration on line 232 and this line on 249 looks like it checks for or indicates that this method would only be called on something where layers exist. Even the check on line 245 makes it seem like passing a layer in is optional.

However, it's highly possible that I simply have overlooked some comment that this particular class is only to be used by things using layers.

Reply:
Or in short, I'm missing what makes this assertion true:
>  If we get here, we know that the session is using layers.

Reply:
> Or in short, I'm missing what makes this assertion true:
> 
> > If we get here, we know that the session is using layers.

'session.renderState.baseLayer' is always false if you are using layers so from line 245 on, we know we handle a layer. line 275 handles the non-layers case.

Reply:
Got it thanks. That's the part I was missing. Consider a comment along those lines